builtin/providers/kubernetes/structures_pod.go

```diff
@@ -0,0 +1,494 @@
+	att := make(map[string]interface{})
+	if in.ActiveDeadlineSeconds != nil {
+		att["active_deadline_seconds"] = *in.ActiveDeadlineSeconds
+	}
+	att["containers"] = flattenContainers(in.Containers)
+
+	att["dns_policy"] = in.DNSPolicy
+
+	att["host_ipc"] = in.HostIPC
+	att["host_network"] = in.HostNetwork
+	att["host_pid"] = in.HostPID
+
+	if in.Hostname != "" {
+		att["hostname"] = in.Hostname
+	}
+	att["image_pull_secrets"] = flattenLocalObjectReferenceArray(in.ImagePullSecrets)
+
+	if in.NodeName != "" {
+		att["node_name"] = in.NodeName
+	}
+	if len(in.NodeSelector) > 0 {
+		att["node_selector"] = in.NodeSelector
+	}
+	if in.RestartPolicy != "" {
+		att["restart_policy"] = in.RestartPolicy
+	}
+
+	if in.SecurityContext != nil {
+		securityContext := flattenPodSecurityContext(in.SecurityContext)
+		if securityContext != nil {
+			att["security_context"] = securityContext
+		}
+	}
+	if in.ServiceAccountName != "" {
+		att["service_account_name"] = in.ServiceAccountName
+	}
+	if in.Subdomain != "" {
+		att["subdomain"] = in.Subdomain
+	}
+
+	if in.TerminationGracePeriodSeconds != nil {
+		att["termination_grace_period_seconds"] = *in.TerminationGracePeriodSeconds
+	}
+
+	volume := userSpec.Volumes
+	if len(in.Volumes) > 0 {
+		att["volumes"] = flattenVolumes(in.Volumes, volume)
+	}
+	return []interface{}{att}
+}
+
+func flattenPodSecurityContext(in *v1.PodSecurityContext) []interface{} {
+	att := make(map[string]interface{})
+	if in.FSGroup != nil {
+		att["fs_group"] = *in.FSGroup
+	}
+
+	if in.RunAsNonRoot != nil {
+		att["run_as_non_root"] = *in.RunAsNonRoot
+	}
+
+	if in.RunAsUser != nil {
+		att["run_as_user"] = *in.RunAsUser
+	}
+
+	if len(in.SupplementalGroups) > 0 {
+		att["supplemental_groups"] = newInt64Set(schema.HashSchema(&schema.Schema{
+			Type: schema.TypeInt,
+		}), in.SupplementalGroups)
+	}
+	if in.SELinuxOptions != nil {
+		att["se_linux_options"] = flattenSeLinuxOptions(in.SELinuxOptions)
+	}
+
+	//K8s is sending back empty context even when none is set during creation
+	//So send nil to the caller of this function to inform that
+	if len(att) == 0 {
+		return nil
+	}
+	return []interface{}{att}
+}
+
+func flattenSeLinuxOptions(in *v1.SELinuxOptions) []interface{} {
+	att := make(map[string]interface{})
+	if in.User != "" {
+		att["user"] = in.User
+	}
+	if in.Role != "" {
+		att["role"] = in.Role
+	}
+	if in.User != "" {
+		att["type"] = in.Type
+	}
+	if in.Level != "" {
+		att["level"] = in.Level
+	}
+	return []interface{}{att}
+}
+
+func flattenVolumes(volumes []v1.Volume, userVolume []v1.Volume) []interface{} {
+
+	userVolumeNames := make(map[string]bool, len(userVolume))
+	for _, v := range userVolume {
+		userVolumeNames[v.Name] = true
+	}
+
+	att := make([]interface{}, len(userVolume))
+	for i, v := range volumes {
+		obj := map[string]interface{}{}
+
+		if !userVolumeNames[v.Name] {
+			//Found a volume which was not provided by user in the tf configuration file and must be internal
```

Comment:
I had this  locally on my laptop last time before I had asked for help :smile:. So I am yet to implement the logic you suggested. This would take care of the _userSpec_ in the above review comment